src/main/java/graphql/util/NodeMultiZipper.java

```diff
@@ -106,8 +107,8 @@ public T getCommonRoot() {
     private NodeZipper<T> moveUp(T parent, List<NodeZipper<T>> sameParent) {
         assertNotEmpty(sameParent, "expected at least one zipper");
 
-        Map<String, List<T>> childrenMap = nodeAdapter.getNamedChildren(parent);
-        Map<String, Integer> indexCorrection = new LinkedHashMap<>();
+        Map<String, List<T>> childrenMap = new LinkedHashMap<>(nodeAdapter.getNamedChildren(parent));
```

Comment:
Does child order matter here?

I dont think it does - it can be a HashMap (micro optimisation)